File: frontend/src/app/model/puntuacion.ts
import { Publicacion } from "./publicacion";
import { Usuario } from "./usuario";


export class Puntuacion {
    
    usuario : Usuario;
    publicacion :  Publicacion;
    constructor (){
        this.usuario = new Usuario();
        this.publicacion=new Publicacion();
    }
    
    id:string;
    punto:string;
    

}

File: backend/src/main/java/com/g4ts/agendaapp/model/Puntuacion.java
package com.g4ts.agendaapp.model;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "Puntuacion")
@Getter @Setter @ToString @Builder
@AllArgsConstructor
@NoArgsConstructor
public class Puntuacion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private Integer punto;

    @OneToOne()
    @JoinColumn(name = "idPublicacion")
    private Publicacion publicacion;

    @OneToOne()
    @JoinColumn(name = "idUsuario")
    private Usuario usuario;

}
